WalletWasabi/Helpers/HttpMessageHelper.cs

```diff
@@ -288,13 +288,19 @@ private static async Task<byte[]> GetDecodedChunkedContentBytesAsync(Stream stre
 
 		private static async Task<byte[]> GetDecodedChunkedContentBytesAsync(Stream stream, HttpRequestContentHeaders requestHeaders, HttpResponseContentHeaders responseHeaders, CancellationToken ctsToken = default)
 		{
-			if (responseHeaders is null && requestHeaders is null)
+			if (responseHeaders is null)
 			{
-				throw new ArgumentException("Response and request headers cannot be both null.");
+				if (requestHeaders is null)
+				{
+					throw new ArgumentException("Response and request headers cannot be both null.");
+				}
 			}
-			if (responseHeaders != null && requestHeaders != null)
+			else
 			{
-				throw new ArgumentException("Either response or request headers has to be null.");
+				if (requestHeaders != null)
+				{
+					throw new ArgumentException("Either response or request headers has to be null.");
+				}
```

Comment:
The original is much better.

Reply:
But there is redundant checks of the state of `responseHeaders` and `requestHeaders` in the original.

Reply:
There is no performance bottleneck here.